src/components/BSelect/BSelect.vue

```diff
@@ -0,0 +1,284 @@
+          role="option"
+          tabindex='-1'
+        > {{ option.text }}
+        </li>
+      </ul>
+    </div>
+  </div>
+</template>
+
+<script>
+
+import { Validator } from 'vee-validate'
+
+const options = []
+
+export default {
+  name: 'b-select',
+  validator: null,
+  props: {
+
+    id: {
+      type: String,
+      required: true
+    },
+    selectOptions: {
+      type: Array
+    },
+    isRequired: {
+      type: Boolean,
+      required: true
+    },
+    selectLabel: {
+      type: String,
+      required: true
+    },
+    initialValue: {
+      type: Object,
+      required: false
+    },
+    placeholder: {
+      type: String,
+      required: false,
+      default: 'Choose Option'
+    }
+  },
+
+  methods: {
+
+    /**
+     * Shows and hides line options for the select list.
+     */
+
+    toggleList () {
+      this.opened = !this.opened
+      if (this.opened) {
+        this.$nextTick( function () {
+          this.setFocus()
+        })
+      } else {
+        if (this.isRequired) {
+          this.validate()
+        }
+        document.querySelector(`#dropdown${this.id}`).focus()
+      }
+    },
+
+    /**
+     * @param {Object} option
+     * Handles the selection of a line and closed the list.
+     */
+
+    selectOption (option) {
+      this.selectedOption = option
+      this.opened = !this.opened
+      if (this.isRequired) {
+        this.validate()
+      }
+      document.querySelector(`#dropdown${this.id}`).focus()
+    },
+
+    /**
+     * Handles validation if required.
+     */
+
+    validate () {
+      this.validator.validate('selectedOption', this.selectedOption)
+    },
+
+    /**
+     * Handles the focus setting for accessibility purposes
+     */
+
+    setFocus () {
+      if (this.selectedOption) {
+        const line = document.querySelector(`#${this.id} #option${this.selectedOption.id}`)
+        line.focus()
+      } else {
+        const line = document.querySelector(`#${this.id} #option${this.options[0].id}`)
+        line.focus()  
+      }
+    },
+
+    /**
+     * Handles the up arrow (38) key press event
+     */
+
+    upHandler () {
+      const target = event.target;
+      if(target.previousElementSibling) {
+        const next = target.previousElementSibling;
+        const first = next.previousElementSibling;
+        if (first) {
+          next.setAttribute("tabindex", "0");
+          target.setAttribute("tabindex", "-1");
+          next.focus();
+        }
+      }
+    },
+
+    /**
+     * Handles the down arrow (40) key press event
+     */
+
+    downHandler () {
+      const target = event.target;
+      if(target.nextElementSibling) {
+        const next = target.nextElementSibling;
+        next.setAttribute("tabindex", "0");
+        target.setAttribute("tabindex", "-1");
+        next.focus();
+      }
+    }, 
+
+    clearErrors() {
+      this.errors.clear();
+    }
+  },
+
+  data () {
+    return {
+      options: this.selectOptions,
```

Comment:
Why are we setting this to the data attribute? Why not use `this.selectOptions` within the component?

Reply:
when I first did this I had no idea what I was doing and was just exploring, I'll update it